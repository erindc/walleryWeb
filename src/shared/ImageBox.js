import React from 'react';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import IconButton from '@material-ui/core/IconButton';
import ThumbUpOutlined from '@material-ui/icons/ThumbUpOutlined';
import Flag from '@material-ui/icons/Flag';
import Payment from '@material-ui/icons/Payment';
import Grid from '@material-ui/core/Grid';
import { withStyles } from '@material-ui/core/styles';
import Badge from '@material-ui/core/Badge';
import PropTypes from 'prop-types';
import { prodEndpoint, devEndpoint } from '../services/uploads';

const styles = theme => ({
  imageCard: {
    height: '350px',
  },
  content: {
    height: '80%'
  },
  button: {
    margin: theme.spacing.unit,
  },
  image: {
    maxHeight: '100%',
    width: '100%'
  }
});

const ImageBox = ({ classes, image, handleLike, handleFlag, handlePurchase }) => {
  return (
    <Grid item xs={12} sm={6} md={4}>
      <Card className={classes.imageCard}>
        <CardContent className={classes.content}>
          <img src={prodEndpoint + image.location} alt='art work' className={classes.image} />
        </CardContent>
        <CardActions>
          <IconButton className={classes.button} aria-label="Like" onClick={handleLike}>
            <Badge badgeContent={image.likes} color="primary">
              <ThumbUpOutlined/>
            </Badge>
          </IconButton>
          <div style={{marginLeft:'auto'}}>
            <IconButton className={classes.button} aria-label="Flag" onClick={handleFlag}>
              <Badge badgeContent={image.flags} color="error">
                <Flag/>
              </Badge>
            </IconButton>
            <IconButton className={classes.button} aria-label="Purchase" onClick={handlePurchase}>
              <Payment/>
            </IconButton>
          </div>
        </CardActions>
      </Card>
    </Grid>
  );
}

ImageBox.propTypes = {
  classes: PropTypes.object.isRequired,
  image: PropTypes.object.isRequired,
  handleLike: PropTypes.func.isRequired,
  handleFlag: PropTypes.func.isRequired,
  handlePurchase: PropTypes.func.isRequired
};

export default withStyles(styles)(ImageBox);