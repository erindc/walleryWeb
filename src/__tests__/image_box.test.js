import ImageBox from '../shared/ImageBox';
import React from 'react'
import { shallow } from 'enzyme'

const like = jest.fn();
const flag = jest.fn();
const purchase = jest.fn();

describe('ImageBox', () => {
  it('renders without crashing', () => {
    const component = shallow(<ImageBox image={{id:'id', user_id:'user_id', image_tag:'tag', purchased:false, likes:0, flags:0, location:'somewhere'}} handleLike={like} handleFlag={flag} handlePurchase={purchase} />)
    expect(component).toMatchSnapshot()
  })
})