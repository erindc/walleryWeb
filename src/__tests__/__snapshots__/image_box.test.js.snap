// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageBox renders without crashing 1`] = `
<ImageBox
  classes={
    Object {
      "button": "ImageBox-button-3",
      "content": "ImageBox-content-2",
      "image": "ImageBox-image-4",
      "imageCard": "ImageBox-imageCard-1",
    }
  }
  handleFlag={[MockFunction]}
  handleLike={[MockFunction]}
  handlePurchase={[MockFunction]}
  image={
    Object {
      "flags": 0,
      "id": "id",
      "image_tag": "tag",
      "likes": 0,
      "location": "somewhere",
      "purchased": false,
      "user_id": "user_id",
    }
  }
/>
`;
